zebra/zebra_evpn_mh.c

```diff
@@ -0,0 +1,2145 @@
+			false /* force_clear_static */);
+}
+
+static void zebra_evpn_es_local_info_clear(struct zebra_evpn_es *es)
+{
+	struct zebra_if *zif;
+
+	if (!(es->flags & ZEBRA_EVPNES_LOCAL))
+		return;
+
+	es->flags &= ~ZEBRA_EVPNES_LOCAL;
+	/* if there any local macs referring to the ES as dest we
+	 * need to clear the static reference on them
+	 */
+	zebra_evpn_es_local_mac_update(es,
+			true /* force_clear_static */);
+
+	/* clear the es from the parent interface */
+	zif = es->zif;
+	zif->es_info.es = NULL;
+	es->zif = NULL;
+
+	/* remove from the ES list */
+	list_delete_node(zmh_info->local_es_list, &es->local_es_listnode);
+
+	/* free up the ES if there is no remote reference */
+	zebra_evpn_es_free(es);
+}
+
+/* Delete an ethernet segment and inform BGP */
+static void zebra_evpn_local_es_del(struct zebra_evpn_es *es)
+{
+	struct zebra_evpn_es_evi *es_evi;
+	struct listnode *node = NULL;
+	struct listnode *nnode = NULL;
+	struct zebra_if *zif;
+
+	if (!CHECK_FLAG(es->flags, ZEBRA_EVPNES_LOCAL))
+		return;
+
+	if (IS_ZEBRA_DEBUG_EVPN_MH_ES) {
+		zif = es->zif;
+		zlog_debug("local es %s del; nhg 0x%x if %s",
+				es->esi_str, es->nhg_id,
+				zif ? zif->ifp->name : "-");
+	}
+
+	/* remove all ES-EVIs associated with the ES */
+	for (ALL_LIST_ELEMENTS(es->es_evi_list, node, nnode, es_evi))
+		zebra_evpn_local_es_evi_do_del(es_evi);
+
+	/* send a del if the ES had been sent to BGP earlier */
+	if (es->flags & ZEBRA_EVPNES_READY_FOR_BGP)
+		zebra_evpn_es_send_del_to_client(es);
+
+	zebra_evpn_es_local_info_clear(es);
+}
+
+/* eval remote info associated with the ES */
+static void zebra_evpn_es_remote_info_re_eval(struct zebra_evpn_es *es)
+{
+	/* if there are remote VTEPs the ES-EVI is classified as "remote" */
+	if (listcount(es->es_vtep_list)) {
+		if (!(es->flags & ZEBRA_EVPNES_REMOTE)) {
+			es->flags |= ZEBRA_EVPNES_REMOTE;
+			if (IS_ZEBRA_DEBUG_EVPN_MH_ES)
+				zlog_debug("remote es %s add; nhg 0x%x",
+						es->esi_str, es->nhg_id);
+		}
+	} else {
+		if (es->flags & ZEBRA_EVPNES_REMOTE) {
+			es->flags &= ~ZEBRA_EVPNES_REMOTE;
+			if (IS_ZEBRA_DEBUG_EVPN_MH_ES)
+				zlog_debug("remote es %s del; nhg 0x%x",
+						es->esi_str, es->nhg_id);
+			zebra_evpn_es_free(es);
+		}
+	}
+}
+
+/* A new local es is created when a local-es-id and sysmac is configured
+ * against an interface.
+ */
+static int zebra_evpn_local_es_update(struct zebra_if *zif, uint32_t lid,
+		struct ethaddr *sysmac)
+{
+	struct zebra_evpn_es *old_es = zif->es_info.es;
+	struct zebra_evpn_es *es;
+	esi_t esi;
+	int offset = 0;
+	int field_bytes = 0;
+
+	/* Complete config of the ES-ID bootstraps the ES */
+	if (!lid || is_zero_mac(sysmac)) {
+		/* if in ES is attached to zif delete it */
+		if (old_es)
+			zebra_evpn_local_es_del(old_es);
+		return 0;
+	}
+
+	/* build 10-byte type-3-ESI -
+	 * Type(1-byte), MAC(6-bytes), ES-LID (3-bytes)
+	 */
+	field_bytes = 1;
+	esi.val[offset] = ESI_TYPE_MAC;
+	offset += field_bytes;
+
+	field_bytes = 6;
```

Comment:
**Non blocking comment**

I think we could use `ETH_ALEN` here

Reply:
Ack